### PR TITLE
Initial infrastructure implementation

### PR DESCRIPTION
**Pull request by marcransome**
Resolves: `DVOP-2585`.

### CODE REVIEW

File: groups/dps/data.tf [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  }
+}
+
+data "aws_ec2_managed_prefix_list" "shared_services_management" {
+  name = "shared-services-management-cidrs"
+}
+
+data "aws_iam_user" "concourse" {
+  user_name = "concourse-platform"
+}
```
Comment (sienkin) [inline groups/dps/data.tf]: This data source doesn't appear to be referenced anywhere. Is it required?

Comment (marcransome) [reply groups/dps/data.tf]: Removed in [4268a89](https://github.com/companieshouse/dps-stack/pull/1/commits/4268a89fdf74c76c1e340bd290ab9b60ea02b876).

File: groups/dps/main.tf [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  required_version = ">= 1.3"
+
+  backend "s3" {}
+
+  required_providers {
+    aws = {
+      version = ">= 5.0, < 6.0"
+      source  = "hashicorp/aws"
+    }
+  }
```
Comment (sienkin) [inline groups/dps/main.tf]: We should explicitly define the `vault` and `cloudinit` providers here for completeness.

E.g.:
```
  required_providers {
    aws = {
      version = ">= 5.0, < 6.0"
      source  = "hashicorp/aws"
    }
    cloudinit = {
      source  = "hashicorp/cloudinit"
      version = ">= 2.3.0, < 3.0.0"
    }
    vault = {
      source  = "hashicorp/vault"
      version = ">= 3.23.0, < 4.0.0"
    }
  }
```

File: groups/dps/main.tf [hunk 3]
```diff
@@ -0,0 +1,16 @@
+terraform {
+  required_version = ">= 1.3"
```
Comment (sienkin) [inline groups/dps/main.tf:2]: May be worth setting an upper bound, such as `< 2.0`

**Review by sienkin — APPROVED** [review]

File: groups/dps/main.tf (hunk 3 above)
Comment (marcransome) [reply groups/dps/main.tf:2]: I believe we may have had a brief discussion about this in recent memory—the upper bound for Terraform versions has been removed across numerous projects (see [chl-tuxedo-stack#80](https://github.com/companieshouse/chl-tuxedo-stack/pull/80), [ois-tuxedo-stack#54](https://github.com/companieshouse/ois-tuxedo-stack/pull/54), [fil-tuxedo-stack#120](https://github.com/companieshouse/fil-tuxedo-stack/pull/120), and [ais-tuxedo-stack#23](https://github.com/companieshouse/ais-tuxedo-stack/pull/23)).

These changes were made on the basis that we control the version via our `terraform-runner` wrapper and its related container image, both locally and in CI/CD. This also allows Dependabot to function despite using a more recent version of Terraform. We may want to consider incorporating this into our best practice guide, coupled with the inclusion of a suitable Dependabot configuration file and the Terraform dependency file (`.terraform.lock.hcl`), to guarantee stability of dependencies across Terraform runs and to maintain dependencies long term.

The lower bound serves as an indication of the minimum required Terraform version in all execution scenarios. Without an upper bound, Dependabot is able to function correctly: we have no control over the version of Terraform provided by Dependabot, which is likely to be ahead of the version that we require in most cases, however by removing the upper version bound—and barring any breaking changes introduced in more recent Terraform versions—Dependabot is able to modify the dependency lock file correctly (by executing commands like `terraform init -upgrade -backend=false`).

An issue has been raised in the [dependabot/dependabot-core](https://github.com/dependabot/dependabot-core) project discussing Terraform versions here: [Support project-specific versions of Terraform](https://github.com/dependabot/dependabot-core/issues/5797).

**Pull request merged**